static/js/recent_topics.js

```diff
@@ -1,10 +1,20 @@
 const render_recent_topics_body = require('../templates/recent_topics_table.hbs');
+const render_recent_topic_row = require('../templates/recent_topic_row.hbs');
 const topics = new Map(); // Key is stream-id:topic.
+let initialized = false;
 
 exports.process_messages = function (messages) {
     for (const msg of messages) {
         exports.process_message(msg);
     }
+    if (!initialized) {
+        // We initialize recent topics once we process
+        // initially fetched messages and then we
+        // do inplace updates for the messages fetched
+        // later on.
+        exports.complete_rerender();
+        initialized = true;
+    }
```

Comment:
I think perhaps we wouldn't need this `initialized` flag if we did the rendering logic for when to do complete vs. not change I suggest below.

Reply:
Yes, removed.